src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -252,47 +242,43 @@ public Consumer<Message<String>> consumeRun() {
-                            return Mono.fromRunnable(() -> securityAnalysisResultService.insert(resultContext.getResultUuid(),
-                                            result,
-                                            result.getPreContingencyResult().getStatus() == LoadFlowResult.ComponentResult.Status.CONVERGED ? SecurityAnalysisStatus.CONVERGED : SecurityAnalysisStatus.DIVERGED))
-                                    .then(Mono.just(result))
-                                    .doFinally(ignored -> {
-                                        long finalNanoTime = System.nanoTime();
-                                        LOGGER.info("Stored in {}s", TimeUnit.NANOSECONDS.toSeconds(finalNanoTime - startTime.getAndSet(finalNanoTime)));
-                                    });
-                        })
-                        .doOnSuccess(result -> {
-                            if (result != null) {  // result available
-                                notificationService.emitAnalysisResultsMessage(resultContext.getResultUuid().toString(), resultContext.getRunContext().getReceiver());
-                                LOGGER.info("Security analysis complete (resultUuid='{}')", resultContext.getResultUuid());
-                            } else {  // result not available : stop computation request
-                                if (cancelComputationRequests.get(resultContext.getResultUuid()) != null) {
-                                    cleanASResultsAndPublishCancel(resultContext.getResultUuid(), cancelComputationRequests.get(resultContext.getResultUuid()).getReceiver());
-                                }
-                            }
-                        })
-                        .onErrorResume(throwable -> {
-                            if (!(throwable instanceof CancellationException)) {
-                                LOGGER.error(FAIL_MESSAGE, throwable);
-                                notificationService.emitFailAnalysisMessage(resultContext.getResultUuid().toString(),
-                                        resultContext.getRunContext().getReceiver(),
-                                        throwable.getMessage());
-                                securityAnalysisResultService.delete(resultContext.getResultUuid());
-                                return Mono.empty();
-                            }
-                            return Mono.empty();
-                        })
-                        .doFinally(s -> {
-                            futures.remove(resultContext.getResultUuid());
-                            cancelComputationRequests.remove(resultContext.getResultUuid());
-                            runRequests.remove(resultContext.getResultUuid());
-                        })
-                        .block();
+                startTime.set(System.nanoTime());
+                try {
```

Comment:
Use the same template as other computations  for try/catch into remove the code smell

Reply:
done